src/main/java/com/google/step/snippet/servlets/SearchServlet.java

```diff
@@ -136,7 +146,18 @@ public void doGet(HttpServletRequest request, HttpServletResponse response)
     } catch (InterruptedException | RejectedExecutionException e) {
       allCards = Collections.emptyList();
     }
-
+    if (!preferredSite.isEmpty()) {
+      List<Card> sortedCards = allCards;
```

Comment:
i don't think a new list of card is needed here, you can just do those operation on `allCards`

Reply:
Doing that could result in a ConcurrentAccessException, which is why I copied the list.